src/server/controller.js

```diff
@@ -1,87 +1,189 @@
 var db = require('../db/schema');
+var bcrypt = require('bcrypt');
+
+var saltRounds = 10;
 
 module.exports = {
   signup: {
     post: function(req, res) {
-      // hash password supplied by user
-      // post username and hashed password to database
-      // set up session
-
-      // res.send('controller signup.post');
+      // Check database for username
+      db.User.findAll({
+        where: { username: req.body.username }
+      })
+        .then(function(users) {
+          // If username is free
+          if (users.length === 0) {
+            // Hash password
+            bcrypt.hash(req.body.password, saltRounds, function(err, hash) {
```

Comment:
Actually, I can't tell why yet, but I get an execution error when I omit the salt. Keeping it in for now.